## Q1. Filter even numbers from a list.

In [ ]:
a = [20,21,22,23,24,25,26,27,28,29,30]
b = filter(lambda x: x%2==0, a)
print(list(b))

[20, 22, 24, 26, 28, 30]


## Q2.  Add 2 lists element wise

In [ ]:
a = [1,2,3]
b = [4,5,6]

c = list(map(lambda x,y : x+y, a,b))
print(c)

[5, 7, 9]


## Q3. Extract number greater than 50

In [ ]:
a = [21,31,41,51,61,71,81,91,22,32,42,52,62,72,82,92]

b = filter(lambda x : x>= 50, a)
print(list(b))

[51, 61, 71, 81, 91, 52, 62, 72, 82, 92]


## Q4. Square of all odd numbers in a list

In [13]:
b = [1,2,3,4,5,6,7,8,9]

d = map(lambda x: x**2, filter(lambda x: x % 2 != 0, b))
print(list(d))

[1, 9, 25, 49, 81]


## Q5. Reverse each element in a list

In [25]:
a = ['sam', 'raj']

b = list(map(lambda x : x[::-1], a))
c = b[::-1]
print(c)

['jar', 'mas']


## Q6.  Sum All Numbers in a List

Write a Python function to sum all the numbers in a list.

Sample List : (8, 2, 3, 0, 7)

Expected Output : 20

In [1]:
a = (8, 2, 3, 0, 7)

def sumlist(a):
    sum = 0
    for i in a:
        sum += i
    print("The sum of numbers in a list: ", sum)

sumlist(a)

The sum of numbers in a list:  20


## Q7. Multiply All Numbers in a List

Write a Python function to multiply all the numbers in a list.

Sample List : (8, 2, 3, -1, 7)

Expected Output : -336

In [2]:
b = (8, 2, 3, -1, 7)

def productlist(a):
    product = 1
    for i in a:
        product *= i
    print("The product of elements in a list: ", product)

productlist(b)

The product of elements in a list:  -336


## Q8 Reverse a String

Write a Python program to reverse a string.

Sample String : "1234abcd"

Expected Output : "dcba4321"


In [6]:
str1 = '1234abcd'

def reverselist(a):
    rev = a[::-1]
    print("Reversed String: ", str(rev))

reverselist(str1)

Reversed String:  dcba4321


## Q9. Factorial of a Number

Write a Python function to calculate the factorial of a number (a non-negative integer). The function accepts the number as an argument.

In [5]:
num = int(input("Enter Number (non-negative): "))

def factorial(a):
    fact = 1
    for i in range(1,a+1):
        fact *= i
    print("The factorial of a given Number: ", fact)

factorial(num)

The factorial of a given Number:  120


## Q10. Check if a Number Falls Within a Given Range

Write a Python function to check whether a number falls within a given range.

In [13]:
num = int(input("Enter Number: "))

def in_range(a):
    if a in range(1,101):
        print("The Given Number is in Range")
    else:
        print("The number is Outside range")

in_range(num)

The Given Number is in Range


## Q11. Count Uppercase and Lowercase Letters in a String

Write a Python function that accepts a string and counts the number of upper and lower case letters.


Sample String : 'The quick Brow Fox'

Expected Output :

No. of Upper case characters : 3

No. of Lower case Characters : 12

In [15]:
str2 = 'The quick Brow Fox'

def cases(a):
    uppercount = 0
    lowercount = 0
    for i in a:
        if i.isupper():
            uppercount += 1
        elif i.islower():
            lowercount += 1
    print("The Number of Upper Cases in a String: ", uppercount)
    print("The Number of Lower Cases in a String: ", lowercount)

cases(str2)


The Number of Upper Cases in a String:  3
The Number of Lower Cases in a String:  12


## 12. Return a New List with Distinct Elements from a List

Write a Python function that takes a list and returns a new list with distinct elements from the first list.

Sample List : [1,2,3,3,3,3,4,5]

Unique List : [1, 2, 3, 4, 5]

In [16]:
lst1 = [1,2,3,3,3,3,4,5]

def distinct_list(a):
    b = []
    for i in lst1:
        if i not in b:
            b.append(i)
    print("The Unique List is: ", b)

distinct_list(lst1)

The Unique List is:  [1, 2, 3, 4, 5]


## Q13. Check Whether a Number is Prime

Write a Python function that takes a number as a parameter and checks whether the number is prime or not.

Note : A prime number (or a prime) is a natural number greater than 1 and that has no positive divisors other than 1 and itself.

In [ ]:
num = int(input("Enter Number: "))

def isprime(a):
    if a < 2:
        print("Not a Prime Number")
        return
    
    for i in range(2, int(a ** 0.5) + 1):
        if a % i == 0:
            print("Not a Prime Number")
            return
    print("Prime Number")

isprime(num)

Prime Number


## Q14. Print Even Numbers from a Given List

Write a Python program to print the even numbers from a given list.

Sample List : [1, 2, 3, 4, 5, 6, 7, 8, 9]

Expected Result : [2, 4, 6, 8]

In [33]:
lst2 = [1, 2, 3, 4, 5, 6, 7, 8, 9]

def listofeven(a):
    even = []
    for i in a:
        if i % 2 == 0:
            even.append(i)
    print("The list of even Numbers: ", even)

listofeven(lst2)

The list of even Numbers:  [2, 4, 6, 8]


## Q15. Check if a Number is Perfect

Write a Python function to check whether a number is "Perfect" or not.

According to Wikipedia : In number theory, a perfect number is a positive integer that is equal to the sum of its proper positive divisors, that is, the sum of its positive divisors excluding the number itself (also known as its aliquot sum). Equivalently, a perfect number is a number that is half the sum of all of its positive divisors (including itself).

Example : The first perfect number is 6, because 1, 2, and 3 are its proper positive divisors, and 1 + 2 + 3 = 6. Equivalently, the number 6 is equal to half the sum of all its positive divisors: ( 1 + 2 + 3 + 6 ) / 2 = 6. The next perfect number is 28 = 1 + 2 + 4 + 7 + 14. This is followed by the perfect numbers 496 and 8128.

In [ ]:
num = int(input("Enter Number: "))

def perfect_number(a):
    total = 0
    for i in range(1, a): 
        if a % i == 0:     
            total += i
    if total == a:
        print("Perfect Number")
    else:
        print("Not a Perfect Number")

perfect_number(num)

Not a Perfect Number


## Q16. Check if a String is a Palindrome

Write a Python function that checks whether a passed string is a palindrome or not.

Note: A palindrome is a word, phrase, or sequence that reads the same backward as forward, e.g., madam or nurses run.

In [ ]:
str3 = 'was it a car or a cat i saw'

# "nurses run"
# "never odd or even"
# "a man a plan a canal panama"
# "madam in eden im adam"
# "was it a car or a cat i saw"

def palindrome_string(a):

    b = a.replace(" ", "")
    rev = b[::-1]    

    if b.lower() == rev.lower():
        print("Palindrome String")
    else:
        print("Not Palindrome String")

palindrome_string(str3)

Palindrome String


## Q17. Create and Print a List of Squares for Numbers 1 to 30

Write a Python function to create and print a list where the values are the squares of numbers between 1 and 30 (both included)

In [53]:
lst3 = [x for x in range(1,31)]

def listofsquares(a):
    b = []
    for i in a:
        b.append(i**2)
    print("List of Sqaure is: ", b)

listofsquares(lst3)

List of Sqaure is:  [1, 4, 9, 16, 25, 36, 49, 64, 81, 100, 121, 144, 169, 196, 225, 256, 289, 324, 361, 400, 441, 484, 529, 576, 625, 676, 729, 784, 841, 900]


## Q18. Lambda Add & Multiply

Write a Python program to create a lambda function that adds 15 to a given number passed in as an argument, also create a lambda function that multiplies argument x with argument y and prints the result.

Sample Output:

25

48

In [56]:
b = lambda x : x + 15
print(b(10))

c = lambda x,y : x*y
print(c(12,4))

25
48


## Q19. Tuple Sorting Lambda

Write a Python program to sort a list of tuples using Lambda.

Original list of tuples:
[('English', 88), ('Science', 90), ('Maths', 97), ('Social sciences', 82)]

Sorting the List of Tuples:
[('Social sciences', 82), ('English', 88), ('Science', 90), ('Maths', 97)]

In [13]:
subject_marks = [('English', 88), ('Science', 90), ('Maths', 97), ('Social sciences', 82)]

sorted_marks = sorted(subject_marks, key=lambda x : x[0], reverse= True)
print(sorted_marks)

[('Social sciences', 82), ('Science', 90), ('Maths', 97), ('English', 88)]


## Q20. Write a Python program to sort a list of dictionaries using Lambda.

Original list of dictionaries :

[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}]

Sorting the List of dictionaries :

[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}]

In [63]:
phone = [{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}]

sorted_phone = sorted(phone, key= lambda x : x['color'])
print(sorted_phone)

[{'make': 'Nokia', 'model': 216, 'color': 'Black'}, {'make': 'Samsung', 'model': 7, 'color': 'Blue'}, {'make': 'Mi Max', 'model': '2', 'color': 'Gold'}]


## Q21. Integer Filter Lambda

Write a Python program to filter a list of integers using Lambda.

Original list of integers:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

Even numbers from the said list:
[2, 4, 6, 8, 10]

Odd numbers from the said list:
[1, 3, 5, 7, 9]

In [70]:
numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

even = list(filter(lambda x : x % 2 == 0, numbers))
print("Even Numbers: ",even)
odd = list(filter(lambda x : x % 2 != 0, numbers))
print("Odd Numbers: ",odd)

Even Numbers:  [2, 4, 6, 8, 10]
Odd Numbers:  [1, 3, 5, 7, 9]


## Q22. Square & Cube Lambda

Write a Python program to square and cube every number in a given list of integers using Lambda.

Original list of integers:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

Square every number of the said list:
[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

Cube every number of the said list:
[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

In [72]:
integers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
print("Integers: ", integers)
square = list(map(lambda x : x**2, integers))
cube = list(map(lambda x : x**3, integers))
print("Square of data: ", square)
print("Cube of data: ", cube)

Integers:  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
Square of data:  [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
Cube of data:  [1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]


## Q23. Integer Filter Lambda

Write a Python program to filter a list of integers using Lambda.

Original list of integers:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

Even numbers from the said list:
[2, 4, 6, 8, 10]

Odd numbers from the said list:
[1, 3, 5, 7, 9]

In [4]:
a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

def filter_even_odd(a):
    even = []
    odd = []
    for i in a:
        if i % 2 == 0:
            even.append(i)
        else:
            odd.append(i)
    print(even)
    print(odd)

filter_even_odd(a)

[2, 4, 6, 8, 10]
[1, 3, 5, 7, 9]


## Q24. Square & Cube Lambda

Write a Python program to square and cube every number in a given list of integers using Lambda.

Original list of integers:
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

Square every number of the said list:
[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

Cube every number of the said list:
[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]

In [6]:
b = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

squarelm = list(map(lambda x : x**2, b))
print(squarelm)
cubelm = list(map(lambda x : x**3, b))
print(cubelm)

[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]
[1, 8, 27, 64, 125, 216, 343, 512, 729, 1000]


## Q25. String Start Check Lambda

Write a Python program to find if a given string starts with a given character using Lambda.

Sample Output:

True
False

In [8]:
starts_with = lambda x : True if x.startswith('P') else False

print(starts_with("Python"))
print(starts_with("Java"))

True
False


## Q26. String Number Checker Lambda

Write a Python program to check whether a given string is a number or not using Lambda.

In [11]:
num_checker = lambda x : True if x.isnumeric() else False
print(num_checker("12"))

True


## Q27. Even/Odd Count Lambda

Write a Python program to count the even and odd numbers in a given array of integers using Lambda. Functional programming eBookPython programming course

Original arrays:
[1, 2, 3, 5, 7, 8, 9, 10]

Number of even numbers in the above array: 3

Number of odd numbers in the above array: 5

In [16]:
c = [1, 2, 3, 5, 7, 8, 9, 10]

odd_count = list(filter(lambda x : x % 2 != 0, c))
even_count = list(filter(lambda x : x % 2 == 0, c))
print("Total Odd numbers: ",len(odd_count))
print("Total Even numbers: ",len(even_count))

Total Odd numbers:  5
Total Even numbers:  3


## Q28. Add Lists with Lambda

Write a Python program to add two given lists using map and lambda.

Original list:
[1, 2, 3]

[4, 5, 6]

Result: after adding two list
[5, 7, 9]

In [18]:
a = [1,2,3]
b = [4,5,6]

c = list(map(lambda x,y : x+y, a,b))
print(c)

[5, 7, 9]


## Q29. Divisible by 19 or 13 Lambda

Write a Python program to find numbers divisible by nineteen or thirteen from a list of numbers using Lambda. Python programming course

Orginal list:
[19, 65, 57, 39, 152, 639, 121, 44, 90, 190]

Numbers of the above list divisible by nineteen or thirteen:
[19, 65, 57, 39, 152, 190]

In [21]:
a = [19, 65, 57, 39, 152, 639, 121, 44, 90, 190]

b = list(filter(lambda x : x % 13 == 0 or x % 19 == 0, a))
print(b)

[19, 65, 57, 39, 152, 190]


## Q30. Palindrome Filter Lambda

Write a Python program to find palindromes in a given list of strings using Lambda.

Orginal list of strings:

['php', 'w3r', 'Python', 'abcd', 'Java', 'aaa']

List of palindromes:

['php', 'aaa']

In [26]:
a = ['php', 'w3r', 'Python', 'abcd', 'Java', 'aaa']

b = list(filter(lambda x : x[0] == x[-1], a))
print(b)

['php', 'aaa']


## Q31. Multiply List Lambda

Write a Python program that multiplies each number in a list with a given number using lambda functions. Print the results.

Original list: [2, 4, 6, 9, 11]

Given number: 2

Result:
4 8 12 18 22

In [27]:
a = [2,4,6,9,11]
n = 2

b = list(map(lambda x : x*n, a))
print(b)

[4, 8, 12, 18, 22]
